its/core-it-support/core-it-plugins/maven-it-plugin-active-collection/src/main/java/org/apache/maven/plugin/coreit/CheckThreadSafetyMojo.java

```diff
@@ -133,21 +133,11 @@ public void run() {
 
         getLog().info("[MAVEN-CORE-IT-LOG] Creating output file " + outputFile);
 
-        OutputStream out = null;
-        try {
-            outputFile.getParentFile().mkdirs();
-            out = new FileOutputStream(outputFile);
+        outputFile.getParentFile().mkdirs();
+        try (OutputStream out = new FileOutputStream(outputFile)) {
             componentProperties.store(out, "MAVEN-CORE-IT-LOG");
         } catch (IOException e) {
-            throw new MojoExecutionException("Output file could not be created: " + outputFile, e);
-        } finally {
-            if (out != null) {
-                try {
-                    out.close();
-                } catch (IOException e) {
-                    // just ignore
-                }
-            }
+            throw new MojoExecutionException(e);
```

Comment:
This is changing the logic as now `IOException` when closing are not ignored anymore,
Is it safe to do so?

Reply:
you mean the absent catch `} catch (IOException e) {` in `L:147`?
This is treaded just the same as before, by the `autocloseable` feature. 

Only change is that we not skip the custom message `"Output file could not be created: " + outputFile`, as considered fluff.
Everything else should remain the same, just without all the boilerplate,

Reply:
just like done in:
- https://github.com/apache/maven/pull/2426
- https://github.com/apache/maven/pull/2424

Reply:
> where the file is closed

please do not forget to mention `flush`, as well. Its just as unnecessary to do so, like all the rest we see here.

Having to `flush` manually is very rare, therefore its just old coding propagating **feature envy**, facing the `autocloseable` feature. 

 @gnodet